src/waveform/waveform.h

```diff
@@ -31,6 +31,13 @@ union WaveformData {
 
 class Waveform {
   public:
+    enum class SaveState {
+        Virgin = 0,
```

Comment:
The name Virgin was proposed by dauscher, and I copy-pasted. Pristine might be a better way to describe it, although Empty is more straightforward and less prone to confusion, so I'll change that